### PR TITLE
[CodeStyle][Ruff][BUAA][G-[225-232]] Fix ruff `RUF005` diagnostic for 8 files in `python/test/legacy_test/`

### PR DESCRIPTION
**Pull request by tlxd**
### PR Category

User Experience

### PR Types

Devs

### Description

修复如下文件中 Ruff 检测出的 `RUF005` 问题：

- test/legacy_test/test_imperative_named_members.py
- test/legacy_test/test_imperative_ocr_attention_model.py
- test/legacy_test/test_index_add_op.py
- test/legacy_test/test_index_select_op.py
- test/legacy_test/test_layers.py
- test/legacy_test/test_lu_op.py
- test/legacy_test/test_lu_unpack_op.py
- test/legacy_test/test_matmul_op_with_head.py

### Related links

- #67116

@gouzil

### CODE REVIEW

File: test/legacy_test/test_index_add_op.py [hunk 1]
```diff
@@ -30,12 +30,13 @@ def compute_index_add_ref(
...
         outer_loop = np.prod(x_shape[:axis]).astype(int)
-        x_reshape = [outer_loop] + list(x_shape[axis:])
+        x_reshape = [outer_loop, *list(x_shape[axis:])]
         x_np_reshape = np.reshape(x_np, tuple(x_reshape))
 
         add_value_reshape = [
-            np.prod(add_value_shape[:axis]).astype(int)
-        ] + list(add_value_shape[axis:])
+            np.prod(add_value_shape[:axis]).astype(int),
+            *list(add_value_shape[axis:]),
```
Comment (gouzil) [inline test/legacy_test/test_index_add_op.py]:
```suggestion
            *add_value_shape[axis:],
```

File: test/legacy_test/test_index_select_op.py [hunk 2]
```diff
@@ -45,7 +45,7 @@ def setUp(self):
         self.inputs = {'X': x_np, 'Index': index_np}
         self.attrs = {'dim': self.dim}
         outer_loop = np.prod(self.x_shape[: self.dim])
-        x_reshape = [outer_loop] + list(self.x_shape[self.dim :])
+        x_reshape = [outer_loop, *list(self.x_shape[self.dim :])]
```
Comment (gouzil) [inline test/legacy_test/test_index_select_op.py]: 同上

File: test/legacy_test/test_index_select_op.py [hunk 3]
```diff
@@ -130,7 +130,7 @@ def setUp(self):
         self.inputs = {'X': convert_float_to_uint16(x_np), 'Index': index_np}
         self.attrs = {'dim': self.dim}
         outer_loop = np.prod(self.x_shape[: self.dim])
-        x_reshape = [outer_loop] + list(self.x_shape[self.dim :])
+        x_reshape = [outer_loop, *list(self.x_shape[self.dim :])]
```
Comment (gouzil) [inline test/legacy_test/test_index_select_op.py]: 同上，下面的就不一一评论了

Comment (tlxd) [reply test/legacy_test/test_index_select_op.py]: 好的好的，我马上修改

**Review by gouzil — APPROVED** [review]

**Pull request merged**